### DISCUSSION TITLE
[Maps] Open Google Maps links in Apple Maps

### DISCUSSION THREAD

**Discussion by mbirth** [mshibanami/redirect-web#34] [Redirect Rules]
* Redirect from: `https://www.google.com/maps?*` (Mode: **Wildcard**)
* Redirect to: `https://maps.apple.com/?$2`
* Application: **Default (Safari)**
* Capture Group Processing: \<empty>
* Excluded URL Patterns: \<empty>
* Examples: `https://www.google.com/maps?hl=en&t=k&ll=41.04522730,-72.32289310&q=United+States` (should show `https://maps.apple.com/?hl....`)

**Comment by mshibanami** [comment]
Hi @mbirth, I tried this rule and it works great!

Yesterday I released a new version of Redirect Web that has a new Library section. Do you mind if I add this rule to it? 😀

**Reply by mbirth** [reply]
@mshibanami I don’t mind at all. 😊

**Reply by mshibanami** [reply]
Thanks! By the way, I noticed that there are 2 more types of the Google Maps links. Also, the combination of the `ll` parameter and the `q` parameter seem to work differently on Apple Maps.

I've addressed all of them as https://github.com/mshibanami/redirect-web/pull/37, and now it's in the library, with a thumbnail. Thanks!!

**Comment by mshibanami** [comment]
Updated as follows:

* Since Apple stopped redirecting `https://maps.apple.com/` to the Maps app and started showing their website instead, I changed the rule title in the library to "Open Google Maps URLs in Apple Maps website."
* I [added](https://github.com/mshibanami/redirect-web/commit/934dc33fea756761ed494f094628fac883736d8f) new rules:
    * "Open Google Maps URLs in Apple Maps (via deeplink)": Only available for mobile devices.
    * "Open Google Maps URLs in Apple Maps (via 'open')": Only available for Mac.
